kernel/arch/dreamcast/hardware/syscalls.c

```diff
@@ -0,0 +1,266 @@
+/* KallistiOS ##version##
+
+   syscalls.c
+
+   Copyright (C) 2024 Andy Barajas
+*/
+
+#include <stdio.h>
+
+#include <dc/syscalls.h>
+#include <arch/memory.h>
+
+/*
+    This module along with syscall_font.s contains functions for accessing 
+    the Dreamcast's system calls.
+*/
+
+/*  
+    Each system call is performed via an indirect vector (VEC_*). This means 
+    that rather than calling a fixed address, a function pointer is fetched 
+    from the fixed address and the call goes through this pointer.
+*/
+#define VEC_SYSINFO       (MEM_AREA_P1_BASE | 0x0C0000B0)
+#define VEC_BIOFONT       (MEM_AREA_P1_BASE | 0x0C0000B4)
+#define VEC_FLASHROM      (MEM_AREA_P1_BASE | 0x0C0000B8)
+#define VEC_MISC_GDROM    (MEM_AREA_P1_BASE | 0x0C0000BC)
+#define VEC_GDROM2        (MEM_AREA_P1_BASE | 0x0C0000C0)
+#define VEC_SYSTEM        (MEM_AREA_P1_BASE | 0x0C0000E0)
+
+/*
+    For each indirect vector there is a number of different functions (FUNC_*) 
+    available.  For the VEC_MISC_GDROM vector we have to also supply a super 
+    function (SUPER_FUNC_*) to have access to functions (FUNC_*) for either
+    MISC or GDROM system calls.
+*/
+
+/* SYSINFO functions */
+#define FUNC_SYSINFO_INIT     0
+#define FUNC_SYSINFO_ICON     2
+#define FUNC_SYSINFO_ID       3
+
+/* ROMFONT functions */
+#define FUNC_ROMFONT_ADDRESS  0
+#define FUNC_ROMFONT_LOCK     1
+#define FUNC_ROMFONT_UNLOCK   2
+
+/* FLASHROM functions */
+#define FUNC_FLASHROM_INFO    0
+#define FUNC_FLASHROM_READ    1
+#define FUNC_FLASHROM_WRITE   2
+#define FUNC_FLASHROM_DELETE  3
+
+/* MISC/GDROM super functions */
+#define SUPER_FUNC_MISC       -1
+#define SUPER_FUNC_GDROM      0
+
+/* MISC functions */
+#define FUNC_MISC_INIT        0
+#define FUNC_MISC_SETVECTOR   1
+
+/* GDROM functions */
+#define FUNC_GDROM_SEND_COMMAND     0
+#define FUNC_GDROM_CHECK_COMMAND    1
+#define FUNC_GDROM_EXEC_SERVER      2
+#define FUNC_GDROM_INIT             3
+#define FUNC_GDROM_DRIVE_STATUS     4
+#define FUNC_GDROM_DMA_CALLBACK     5
+#define FUNC_GDROM_DMA_TRANSFER     6
+#define FUNC_GDROM_DMA_CHECK        7
+#define FUNC_GDROM_ABORT_COMMAND    8
+#define FUNC_GDROM_RESET            9
+#define FUNC_GDROM_SECTOR_MODE      10
+#define FUNC_GDROM_PIO_CALLBACK     11
+#define FUNC_GDROM_PIO_TRANSFER     12
+#define FUNC_GDROM_PIO_CHECK        13
+#define FUNC_GDROM_UNKNOWN1         14
+#define FUNC_GDROM_CHANGE_CD        15
```

Comment:
You grab CHANGE_CD from ISO Loader firmware?
Original BIOS doesn't have this syscall, it's for DreamShell only for multi disc GDI images.

Reply:
@DC-SWAT Oh I know that there was 16 FUNC _GDROM.  I can rename to UNKNOWN2.  Would that be okay?

Reply:
They are not unknown, they are just stubs/reserved.